File: dotnet/src/Botas/TeamsConversation.cs
using System.Text.Json.Serialization;

namespace Botas;

public class TeamsConversation : Conversation
{
    [JsonPropertyName("conversationType")] public string? ConversationType { get; set; }
    [JsonPropertyName("tenantId")] public string? TenantId { get; set; }
    [JsonPropertyName("isGroup")] public bool? IsGroup { get; set; }
    [JsonPropertyName("name")] public string? Name { get; set; }
}

File: python/packages/botas/src/botas/teams_conversation.py
from __future__ import annotations

from pydantic import ConfigDict
from pydantic.alias_generators import to_camel

from botas.core_activity import Conversation


class TeamsConversation(Conversation):
    model_config = ConfigDict(
        alias_generator=to_camel,
        populate_by_name=True,
        extra="allow",
    )

    conversation_type: str | None = None
    tenant_id: str | None = None
    is_group: bool | None = None
    name: str | None = None
